src/tools/rbd/action/Export.cc

```diff
@@ -369,6 +376,10 @@ class C_Export : public Context
       }
     }
 
+    if (m_win32_phys_disk) {
+      // Need to ensure that the writes are sector aligned
+      m_bufferlist.rebuild();
```

Comment:
I see, one idea might be compare the sizes and fill the rest of the disk with zeros.

> Which brings me to note that exporting to a block device isn't directly supported on Linux either -- but one can do e.g. rbd export --export-format 1 myimage - > /dev/sda. Does PowerShell (?) provide something similar?

Unfortunately cmd and powershell do not allow redirecting the output to a block device.

Reply:
> I see, one idea might be compare the sizes and fill the rest of the disk with zeros.

I think you can just disable that check for physical disks, just like it's already disabled for `stdout`.

> Unfortunately cmd and powershell do not allow redirecting the output to a block device.

I think it would be nice to extend this physical disk support to Linux as well, for uniformity and also better test coverage if nothing else.

Reply:
I've extended this to Linux as well. I'm marking it as a draft until I get to verify it and maybe add some tests as well.

Also added some error messages for the unsupported operations: v2 imports/exports using block devices as well as diff imports/exports.